depends/funcs.mk

```diff
@@ -186,7 +186,7 @@ $($(1)_preprocessed): | $($(1)_extracted)
 	$(AT)$(foreach patch,$($(1)_patches),cd $(PATCHES_PATH)/$(1); cp $(patch) $($(1)_patch_dir) ;)
 	$(AT)cd $$(@D); $(call $(1)_preprocess_cmds, $(1))
 	$(AT)touch $$@
-$($(1)_configured): | $($(1)_dependencies) $($(1)_preprocessed)
+$($(1)_configured): | $($(1)_all_dependencies) $($(1)_preprocessed)
```

Comment:
What does this fix?

Reply:
`$($(1)_dependencies)` doesn't include `$($($(1)_type)_native_toolchain) $($($(1)_type)_native_binutils)`, meaning that:
```sh
$ make -C depends/ HOST=x86_64-apple-darwin16 print-zlib_dependencies
make: Entering directory '/home/dongcarl/src/bitcoin/master/depends'
zlib_dependencies =
make: Leaving directory '/home/dongcarl/src/bitcoin/master/depends'
```
while
```
$ make -C depends/ HOST=x86_64-apple-darwin16 print-zlib_all_dependencies
make: Entering directory '/home/dongcarl/src/bitcoin/master/depends'
zlib_all_dependencies = native_cctools
make: Leaving directory '/home/dongcarl/src/bitcoin/master/depends'
```

This fixes: https://github.com/bitcoin/bitcoin/issues/19799

I'd be happy to leave this for another PR, and figure out what the right fix is.

Reply:
Please see an alternative #19868.